lib/cdo/high_frequency_reporter.rb

```diff
@@ -0,0 +1,76 @@
+require 'csv'
+require 'time'
+# Usage:
+#   errors = HighFrequencyReporter.new(Slack.new)
+#   errors.load!
+#   loop do
+#     <stuff>
+#     errors.record "Something went wrong"
+#   end
+#   errors.report!
+#   errors.save
+class HighFrequencyReporter
+  attr_accessor :new_events
+
+  def initialize(chat_client, channel, log_file_name)
+    @chat_client = chat_client
+    @channel = channel
+    # Load old events from disk
+    # [{name: "eventblah", reported_at: '2020010...'}, ...]
+    @old_events = []
+    @new_events = []
+    @filename = log_file_name
+  end
+
+  # Loads known events from previous runs from a file on disk
+  def load
```

Comment:
Returns a true/false value so the caller knows if this method succeeds or not. Also, adds a comment that this method swallows exceptions.

Reply:
Been neglecting this for a month :) Just pushed some changes to get at some of your comments, take a look when you have a minute!